packages/dapp/src/components/listing/CommitVoteDetail.tsx

```diff
@@ -0,0 +1,155 @@
+
+export interface CommitVoteDetailProps {
+  challengeID: BigNumber;
+}
+
+export interface CommitVoteDetailState {
+  isVoteTokenAmtValid: boolean;
+  voteOption?: string;
+  salt?: string;
+  numTokens?: string;
+}
+
+export interface InputElementProps {
+  type: string;
+  name?: string;
+  value?: string;
+  readOnly?: boolean;
+  onChange?(event: any): void;
+  validate?(event: any): boolean;
+}
+
+class InputElement extends React.Component<InputElementProps> {
+  constructor(props: any) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    return (
+      <input
+        type={this.props.type}
+        name={this.props.name}
+        value={this.props.value}
+        readOnly={this.props.readOnly}
+        onChange={this.onChange.bind(this)}
+      />
+    );
+  }
+
+  private onChange(event: any): void {
+    if (this.props.onChange) {
+      this.props.onChange(event);
+    }
+
+    if (this.props.validate) {
+      this.props.validate(event);
+    }
+  }
+}
+
+class CommitVoteDetail extends React.Component<CommitVoteDetailProps, CommitVoteDetailState> {
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      isVoteTokenAmtValid: false,
+    };
+
+    this.updateCommitVoteParam = this.updateCommitVoteParam.bind(this);
+  }
+
+  public render(): JSX.Element {
+    return (
+      <StyledFormContainer>
+        <h3>Vote On Challenge</h3>
+
+        {/* @TODO/jon: Change this to a hidden form field.
+            The typical user/voter doesn't need to know about the PollID */}
+        <FormGroup>
+          <label>Poll ID</label>
+          <InputElement type="text" name="" value={this.props.challengeID.toString()} readOnly={true} />
+        </FormGroup>
+
+        <FormGroup>
+          <label>Support This Challenge (Vote Option)</label>
+          <InputElement type="radio" value="1" name="voteOption" onChange={this.updateCommitVoteParam} /> Yes
+          <InputElement type="radio" value="0" name="voteOption" onChange={this.updateCommitVoteParam} /> No
+        </FormGroup>
+
+        <FormGroup>
+          <label>Salt</label>
+          <InputElement type="text" name="salt" onChange={this.updateCommitVoteParam} />
+        </FormGroup>
+
+        <FormGroup>
+          <label>Number of Tokens</label>
+          {!this.state.isVoteTokenAmtValid && <FormValidationMessage children="Please enter a valid token amount" />}
+          <InputElement
+            type="text"
+            name="numTokens"
+            validate={this.validateVoteCommittedTokens.bind(this)}
+            onChange={this.updateCommitVoteParam}
+          />
+        </FormGroup>
+
+        <FormGroup>
+          <TransactionButton
+            firstTransaction={this.requestVotingRights}
+            secondTransaction={this.commitVoteOnChallenge}
+          >
+            Commit Vote
+          </TransactionButton>
+        </FormGroup>
+      </StyledFormContainer>
+    );
+  }
+
+  private updateCommitVoteParam(event: any): void {
```

Comment:
if you change this to `updateCommitVoteParam = (event:any): void => {` you can remove the `bind` in the constructor